### PR TITLE
Fix: Nested modal

### PR DESCRIPTION
**Pull request by AAbosham**
Hi about this issue #4888 `wire:key` will solved it

### CODE REVIEW

**Comment by zepfietje** [comment]
I'm pretty sure this will cause other issues. @danharrin?

**Comment by danharrin** [comment]
Modals have `wire:key`s already - but passed through `$attributes`. Since the `wire:key` should be a different structure depending on the action.

And yes a generic wire:key based on the modal ID would reintroduce issues we've fixed relating to form components within the modals.

**Comment by danharrin** [comment]
I guess we could add the Livewire ID to the key and it would be ok

File: packages/support/resources/views/components/modal/index.blade.php [hunk 1]
```diff
@@ -54,6 +54,7 @@
     aria-modal="true"
     class="filament-modal {{ $displayClasses }}"
     wire:ignore.self
+    wire:key="{{ $id }}-modal"
```
Comment (danharrin) [inline packages/support/resources/views/components/modal/index.blade.php]:
```suggestion
    wire:key="{{ $this->id }}-{{ $id }}-modal-container"
```

Comment (danharrin) [reply packages/support/resources/views/components/modal/index.blade.php]: Please commit this and I will merge

File: packages/support/resources/views/components/modal/index.blade.php [hunk 2]
```diff
@@ -54,6 +54,7 @@
     aria-modal="true"
     class="filament-modal {{ $displayClasses }}"
     wire:ignore.self
+    wire:key="{{ $this->id }}-{{ $id }}-modal-container"
```
Comment (zepfietje) [inline packages/support/resources/views/components/modal/index.blade.php:57]: @danharrin don't we use `.` instead of `-` in other places?

Comment (danharrin) [reply packages/support/resources/views/components/modal/index.blade.php:57]: yeah i fixed it the minute after i merged lol

Comment (zepfietje) [reply packages/support/resources/views/components/modal/index.blade.php:57]: Lol, didn't see that commit. Thanks.

**Pull request merged**